

# In this project, the dataset named Telco Customer Churn from Kaggle was used. Each row represents a customer, each column contains the customer’s attributes. This dataset contains 21 columns (variables) and 7043 rows (customers) with information such as customerID, gender, Phone Service, and Internet Service.

# Analysis of data columns to identify independent and dependent variables:

# X is the independent variables — the variables we are using to make predictions

- customerID — unique value identifying customer
- gender — whether the customer is a male or a female
- SeniorCitizen — whether the customer is a senior citizen or not (1, 0)
- Partner — whether the customer has a partner or not (Yes, No)
- Dependents — whether the customer has dependents or not (Yes, No). A dependent is a person who relies on another as a primary source of income,
- tenure — number of months the customer has stayed with the company
- PhoneService — whether the customer has a phone service or not (Yes, No)
- MultipleLines — whether the customer has multiple lines or not (Yes, No, No phone service)
- InternetService — customer’s internet service provider (DSL, Fiber optic, No)
- OnlineSecurity — whether the customer has online security or not (Yes, No, No internet service)
- OnlineBackup — whether the customer has online backup or not (Yes, No, No internet service)
- DeviceProtection — whether the customer has device protection or not (Yes, No, No internet service)
- TechSupport — whether the customer has tech support or not (Yes, No, No internet service)
- StreamingTV — whether the customer has streaming TV or not (Yes, No, No internet service)
- StreamingMovies — whether the customer has streaming movies or not (Yes, No, No internet service)
- Contract — type of contract according to duration (Month-to-month, One year, Two year)
- PaperlessBilling — bills issued in paperless form (Yes, No)
- PaymentMethod — payment method used by customer (Electronic check, Mailed check, Credit card (automatic), Bank transfer (automatic))
- MonthlyCharges — amount of charge for service on monthly bases
- TotalCharges — cumulative charges for service during subscription (tenure) period
# y is dependent variable — variable we are trying to predict or estimate

# Churn — output value, predict variable

# So, first of all we imported the required libraries needed 

In [1]:
# Data processing
import pandas as pd
import numpy as np

# Data Visualization
import matplotlib.pyplot as plt
import seaborn as sns

c:\ProgramData\anaconda4\Lib\site-packages\pandas\core\arrays\masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).
  from pandas.core import (


# Next, after installing the  the required packages ,Data Reading. Let’s import the dataset.

- The first step of the analysis consists of reading and storing the data in a Pandas data frame using the pandas.read_csv function we loaded the dataframe, by reading the csv file provided 

In [9]:
df = pd.read_csv("telco_customer_churn.csv")

# After reading the csv as a df, we went ahead to check the first 5 rows  from the top . This returned teh first 5 rpws with the names of cthe columns in addition to the specified rows

In [10]:
df.head(4)

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   

               PaymentMethod MonthlyCharges  TotalCharges Churn  
0           Electronic check          29.85         29.85    No  
1               Mailed check          56.95        1889.5    No  
2               Mailed check          53.85        108.15   Yes  
3  Bank transfer (automatic)          42.30       1840.75    No  

[4 rows x 21 columns]

# Next, we checked the shape of our datafraem uisng .shape

In [13]:
df.shape

(7043, 21)

- We found out that the shape of our dataframe is (7043, 21), this means that the datafrem has 7043 rows and 21 columns

# Show All of The Columns

In [16]:
df.columns.values

array(['customerID', 'gender', 'SeniorCitizen', 'Partner', 'Dependents',
       'tenure', 'PhoneService', 'MultipleLines', 'InternetService',
       'OnlineSecurity', 'OnlineBackup', 'DeviceProtection',
       'TechSupport', 'StreamingTV', 'StreamingMovies', 'Contract',
       'PaperlessBilling', 'PaymentMethod', 'MonthlyCharges',
       'TotalCharges', 'Churn'], dtype=object)

- So, here we can see all of the columns name and immediately we can see some interesting columns like customer Id, gender, phone service, internet service, contract, monthly charges, tenure and obviously churn

# Next we checked for the info for our df so as we can get a coincise summary of the dataframe

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 


- After checking the info of our dataframe, we have 7043 entries and 21 columns. Also, df.info() gave us the nammes of our columns that we have in the df and the data types of each columns. Also,  the number of non null values wa disp;ayed for eah column as shown.
- So our datfarme has three datatypes( floats for 1 column which si monthly charges, int 64 for two columns which include seniour citizena  and tenure and the  other 18 columns are have object datatype)

# Next step  involved cHecking for the decription of our data   in the df using decribe()

In [5]:
df.describe()

SeniorCitizen       tenure  MonthlyCharges
count    7043.000000  7043.000000     7043.000000
mean        0.162147    32.371149       64.761692
std         0.368612    24.559481       30.090047
min         0.000000     0.000000       18.250000
25%         0.000000     9.000000       35.500000
50%         0.000000    29.000000       70.350000
75%         0.000000    55.000000       89.850000
max         1.000000    72.000000      118.750000

- From the summmary we can see that teh decribe function did apply on the three columns since they  are numerical.
- So first we get the  count( which is the number of non empty values) of the senior citizenshop, tenure and monthly charges to be 7034.
- Next the decribe provides  the mean value( seniorcitizen has a mean value of 0.162147, tenure has  has a mean of 32.371149 and Montlycharges has 64.76)
- Next is standard deviation of the three numerical columns( Seniorcitizen = 0.36, tenure = 24.559 and Monthlycharges = 30.09)
- Alos, the summry provides the min value ans max value on each of the columns. Therefore for Senior Citizen it is clear that the min value is 0 while the max value is 1, for tenure the min val is 0 and max value is 72.0 while for monthlycharges the min val is 18.25 while the max value is 118.75
- Lastly the decribe summary gave us the 25th, 5Oth and 75th percentile. The 25th percentile is 0, 9 and 35.5 repectively. The 75% is 0, 55 and 89.85 respectively. The 50th percentiile is also the median which is 0, 29 and 70.35 repectively


# Next we checked for any npn null values in the dataframe 

In [14]:
df.isnull().sum()

customerID          0
gender              0
SeniorCitizen       0
Partner             0
Dependents          0
tenure              0
PhoneService        0
MultipleLines       0
InternetService     0
OnlineSecurity      0
OnlineBackup        0
DeviceProtection    0
TechSupport         0
StreamingTV         0
StreamingMovies     0
Contract            0
PaperlessBilling    0
PaymentMethod       0
MonthlyCharges      0
TotalCharges        0
Churn               0
dtype: int64

- After running the cell we can see the columns name in the left and number for missing values for each column on the right and right every single value is zero so this tells us that this data has no missing values.

# Next we checked to see if we have any duplicated values in out dataframe

In [15]:
df[df.duplicated()]

Empty DataFrame
Columns: [customerID, gender, SeniorCitizen, Partner, Dependents, tenure, PhoneService, MultipleLines, InternetService, OnlineSecurity, OnlineBackup, DeviceProtection, TechSupport, StreamingTV, StreamingMovies, Contract, PaperlessBilling, PaymentMethod, MonthlyCharges, TotalCharges, Churn]
Index: []

[0 rows x 21 columns]

In [7]:
df = df.drop_duplicates()